apps/app/pages/admin/lectures/presences/index.tsx

```diff
@@ -0,0 +1,288 @@
+import { useEffect, useState } from "react";
+import { useRouter } from "next/router";
+import { Button, Row, Select, Space, Table, Typography } from "antd";
+import { CloseOutlined, EditOutlined, SaveOutlined } from "@ant-design/icons";
+
+import {
+  getNinjaEvents,
+  listEvents,
+  listLectures,
+  listMentors,
+} from "bokkenjs";
+import * as api from "bokkenjs";
+import { notifyError, notifyInfo } from "~/components/Notification";
+import AppLayout from "~/layouts/AppLayout";
+import { set } from "lodash-es";
+const { Title } = Typography;
+
+type Event = {
+  id: string;
+  title: string;
+  location_id: string;
+};
+
+type Lecture = {
+  id: string;
+  event: Event;
+  mentor: {
+    id: string;
+    first_name: string;
+    last_name: string;
+    photo: string;
+  };
+  ninja: {
+    id: string;
+    first_name: string;
+    last_name: string;
+    photo: string;
+  };
+  summary: string;
+  notes: string;
+  attendance: string;
+};
```

Comment:
This type definitions are more generic than this page. I think we can move them to the `bokkenjs` package.